VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -289,8 +289,12 @@ public Fragment getItem(int pos) {
             CourseComponent unit = getUnit(pos);
             CourseUnitFragment unitFragment;
             //FIXME - for the video, let's ignore studentViewMultiDevice for now
-            if (unit instanceof VideoBlockModel && ((VideoBlockModel)unit).getData().encodedVideos.getPreferredVideoInfo() != null) {
-                unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
+            if (unit instanceof VideoBlockModel &&
+                    ((VideoBlockModel) unit).getData().encodedVideos.getPreferredVideoInfo() != null) {
+                    unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
+            } else if (unit instanceof VideoBlockModel &&
```

Comment:
**If the youtube url is always there as the fallback**
Instead of having the `instanceof` check twice, it can be simplified into a parent `if` with a child `if-else` like:
```java
if (unit instanceof VideoBlockModel) {
    if (((VideoBlockModel) unit).getData().encodedVideos.getPreferredVideoInfo() != null) {
        unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
    } else {
        unitFragment = CourseUnitOnlyOnYoutubeFragment.newInstance(unit);
    }
}
```